src/lib/yang/schema.lua

```diff
@@ -794,14 +794,15 @@ function resolve(schema, features)
          end
       end
       if node.body then
-         node.body = shallow_copy(node.body)
-         for k,v in pairs(node.body or {}) do
+         local body = node.body or {}
+         node.body = {}
+         for k,v in pairs(body) do
             if v.kind == 'uses' then
                -- Inline "grouping" into "uses".
                local grouping = lookup_lazy(env, 'groupings', v.id)
-               node.body[k] = nil
+               body[k] = nil
```

Comment:
true, it's not needed. Still works though because removing elements in a map seems to be OK but what's not OK is to add new keys while iterating.